### PR TITLE
hotfix: evo2 divergence - downgrade TE to v1.13

### PR DESCRIPTION
**Pull request by dorotat-nv**
### Description
We have observed a divergence on ToT main for Evo2 1b 8k  pretrainign. We suspect that the reason for it is change of pytorch container (from 24.12 to 25.01) or TransformerEngine version (from v1.13 to v1.14). Hence we run the same evo2 training on dockers build based on different TE and pytorch assumptions. Below are the 

The above experiments result in no divergence for
* pytorch 25.01-py3 (default [TE_TG=v1.14](https://docs.nvidia.com/deeplearning/frameworks/pytorch-release-notes/rel-25-01.html)), with hardcoded TE_TAG=v1.13 and
* pytorch 24.12-py3 (by default [TE_TG=v1.13](https://docs.nvidia.com/deeplearning/frameworks/pytorch-release-notes/rel-24-12.html))

but with the divergence for
* (ToT main) pytorch 25.01-py3 (default [TE_TG=v1.14](https://docs.nvidia.com/deeplearning/frameworks/pytorch-release-notes/rel-25-01.html))
* pytorch 25.02-py3 (by default [TE_TG=v2.0](https://docs.nvidia.com/deeplearning/frameworks/pytorch-release-notes/rel-25-02.html))
### Type of changes

- [x]  Bug fix (non-breaking change which fixes an issue)
- [ ]  New feature (non-breaking change which adds functionality)
- [ ]  Refactor
- [ ]  Documentation update
- [ ]  Other (please describe):

### CI Pipeline Configuration
Configure CI behavior by applying the relevant labels:

- [SKIP_CI](https://github.com/NVIDIA/bionemo-framework/blob/main/docs/docs/user-guide/contributing/contributing.md#skip_ci) - Skip all continuous integration tests
- [INCLUDE_NOTEBOOKS_TESTS](https://github.com/NVIDIA/bionemo-framework/blob/main/docs/docs/user-guide/contributing/contributing.md#include_notebooks_tests) - Execute notebook validation tests in pytest
- [INCLUDE_SLOW_TESTS](https://github.com/NVIDIA/bionemo-framework/blob/main/docs/docs/user-guide/contributing/contributing.md#include_slow_tests) - Execute tests labelled as slow in pytest for extensive testing

> [!NOTE]
> By default, the notebooks validation tests are skipped unless explicitly enabled.

#### Authorizing CI Runs

We use [copy-pr-bot](https://docs.gha-runners.nvidia.com/apps/copy-pr-bot/#automation) to manage authorization of CI
runs on NVIDIA's compute resources.

* If a pull request is opened by a trusted user and contains only trusted changes, the pull request's code will
  automatically be copied to a pull-request/ prefixed branch in the source repository (e.g. pull-request/123)
* If a pull request is opened by an untrusted user or contains untrusted changes, an NVIDIA org member must leave an
  `/ok to test` comment on the pull request to trigger CI. This will need to be done for each new commit.

### Usage

```python
TODO: Add code snippet
```

### Pre-submit Checklist

 - [ ] I have tested these changes locally
 - [ ] I have updated the documentation accordingly
 - [ ] I have added/updated tests as needed
 - [ ] All existing tests pass successfully

### CODE REVIEW

**Review by jstjohn — APPROVED** [review]
Thank you!

**Review by pstjohn — APPROVED** [review]

**Review by trvachov — CHANGES_REQUESTED** [review]
We cannot downgrade. Blackwell will stop working.

**Comment by trvachov** [comment]
@dorotat-nv please revert
